internal/services/servicenetworking/application_load_balancer_association_resource.go

```diff
@@ -0,0 +1,228 @@
+			trafficControllerClient := metadata.Client.ServiceNetworking.TrafficControllerInterface
+			client := metadata.Client.ServiceNetworking.AssociationsInterface
+
+			var config AssociationModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding %v", err)
+			}
+
+			albId, err := trafficcontrollerinterface.ParseTrafficControllerID(config.ApplicationLoadBalancerId)
+			if err != nil {
+				return err
+			}
+
+			id := associationsinterface.NewAssociationID(albId.SubscriptionId, albId.ResourceGroupName, albId.TrafficControllerName, config.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of exisiting %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(t.ResourceType(), id)
+			}
+
+			controller, err := trafficControllerClient.Get(ctx, *albId)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *albId, err)
+			}
+
+			association := associationsinterface.Association{
+				Properties: &associationsinterface.AssociationProperties{
+					Subnet: &associationsinterface.AssociationSubnet{
+						Id: config.SubnetId,
+					},
+					AssociationType: associationsinterface.AssociationTypeSubnets,
+				},
+			}
+
+			if controller.Model == nil {
+				return fmt.Errorf("retrieving %s: model was nil", *albId)
+			}
+
+			association.Location = location.Normalize(controller.Model.Location)
+
+			if len(config.Tags) > 0 {
+				association.Tags = &config.Tags
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, association); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (t ApplicationLoadBalancerAssociationResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ServiceNetworking.AssociationsInterface
+
+			id, err := associationsinterface.ParseAssociationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retreiving %s: %v", id.ID(), err)
+			}
+
+			trafficControllerId := associationsinterface.NewTrafficControllerID(id.SubscriptionId, id.ResourceGroupName, id.TrafficControllerName)
+			state := AssociationModel{
+				Name:                      id.AssociationName,
+				ApplicationLoadBalancerId: trafficControllerId.ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				state.Tags = pointer.From(model.Tags)
+
+				if prop := model.Properties; prop != nil {
+					if prop.Subnet != nil {
+						parsedSubnetId, err := commonids.ParseSubnetID(prop.Subnet.Id)
```

Comment:
When parsing IDs in the Read we need to parse these insensitively:

```suggestion
						parsedSubnetId, err := commonids.ParseSubnetIDInsensitively(prop.Subnet.Id)
```